internal/backend/dispatcher/eventsworkflow.go

```diff
@@ -181,14 +181,36 @@ func (d *dispatcher) signalWorkflows(ctx context.Context, event sdktypes.Event)
 		return nil
 	}
 
-	signals, err := d.db.ListSignalsWaitingOnConnection(ctx, conn.ID(), event.Type())
+	signals, err := d.db.ListSignalsWaitingOnConnection(ctx, conn.ID())
 	if err != nil {
 		z.Error("could not fetch signals", zap.Error(err))
 		return err
 	}
 
 	z.Debug("found signal candidates", zap.Int("count", len(signals)))
 	for _, signal := range signals {
+		match, err := event.Matches(signal.Filter)
+		if err != nil {
+			z.Error("inavlid signal filter",
```

Comment:
i wonder if it's an error, since it's actually a user error, not a backend error. can you add a todo with a ticket to notify the user of these somehow?

Reply:
ok